src/pybullet_industrial/g_code_processor.py

```
@@ -0,0 +1,435 @@
+                    self.elementary_operations.append(lambda: operation())
+            else:
+                path = self.build_path()
+                self.elementary_operations = self.create_movement_operations(
+                    path)
+
+        elif m_int is not None:
+            for operation in self.m_commands[str(m_int)]:
+                self.elementary_operations.append(lambda: operation())
+
+        elif t_int is not None:
+            for operation in self.t_commands[str(t_int)]:
+                self.elementary_operations.append(lambda: operation())
+
+            self.elementary_operations.append(lambda: self.calibrate_tool())
+
+    def build_path(self):
+        """Calculates a new point and new orientation based on the new
+        coordinates and offset. Depending on the G-command type, a
+        specific tool path is returned. G0 commands create a path
+        with 2 interpolation steps, while higher G-1-2-3
+        commands generate a path with the chosen precision.
+
+        Returns:
+            path(ToolPath): Interpolated tool path
+        """
+
+        cmd = self.gcode[self.index_gcode]
+        g_com = cmd[0][1]
+
+        self.build_new_point(cmd)
+
+        if g_com == 0:
+            path = self.build_simple_path()
+        else:
+            path = self.build_precise_path(g_com)
+
+        orientation = p.getQuaternionFromEuler(self.new_or)
+        path.orientations = np.transpose([orientation]
+                                         * len(path.orientations[0]))
+
+        return path
+
+    def build_new_point(self, cmd: list):
+        """Calculates the new point of a G-Code command with respect
+        to the current offset.
+
+        Args:
+            cmd(list): Current command line
+        """
+
+        variables = {'X': np.nan, 'Y': np.nan, 'Z': np.nan, 'A': np.nan,
+                     'B': np.nan, 'C': np.nan, 'R': np.nan}
+
+        for val in cmd:
+            if val[0] in variables:
```

and if not we might want to throw an error

---

error for unknown letter adapted